chore: Quiet numerous DeepSource reported issues

rasa opened the pull request:
### Purpose

Quiet DeepSource issues reported in
https://deepsource.io/gh/syncthing/syncthing/run/1f029f7e-c9e4-40a1-9b59-d1a8a4d39f1f/go/

### Testing

All tests pass.

### Comments

I chose to quiet things like
```
func (fs *errorFilesystem) Chmod(name string, mode FileMode) error { return fs.err }
```
by prefixing it with:
```
// skipcq: RVV-B0012 : parameter 'name' seems to be unused, consider removing or renaming it as _
```
as I didn't want to rewrite someone else's code as:
```
func (fs *errorFilesystem) Chmod(_ string, _ FileMode) error { return fs.err }
```
without direction.

calmh reviewed (commented):
Whole I appreciate the effort you've put in to write this, I don't think we should add these magic comments. In my opinion they're ugly, and static checkers come and go. We've used several previously that are no longer a thing, and probably deep source will wander the same path. At that point I don't want hundreds of cryptic incantations to it "littered" all over the code.

I also think it's fine to let it complain about things that are just it's opinion, or not applicable in the condition it complains about it. The intention is for a human to consider it and draw a conclusion one way or the other.

rasa commented:
Understood. I will remove the comments from the PR, which will require code changes to quiet some of the issues.

While I appreciate your opinion, please consider:
1. the comments indicate a human has reviewed the item and determined it's not an issue.
1. having DeepSource saying most PRs are broken ( ❌ ) with sometimes hundreds of reported issues will discourage us from bothering to review its reports, as 99% of the time it's just noise, that we'll assume someone else has already reviewed.
1. wading thru pages of the same errors over and over again is not conducive to actually finding issues. Most users will simply not bother.
1. the comments can be stripped out with a single: `sed -i -Ee '|^\s*//\s*skipcq:|d; s|//\s*skipcq:.*||'`.
1. the comments can be added at the end of the line instead.

I couldn't find anything that allows us to disable reporting specific issues in the .deepsource.toml file. Most other linters have this feature, so maybe this functionality exists, but isn't well documented.

AudriusButkevicius commented:
Skimmed some of the comments, we should just fix the code where it makes sense, ie, not to shadow built ins, rename things to _ etc, oppose to adding comments to silence it.

rasa commented:
> Skimmed some of the comments, we should just fix the code where it makes sense, ie, not to shadow built ins, rename things to _ etc, oppose to adding comments to silence it.
> 
> And ignore the ones that make no sense.

Agreed, thanks. 

If I find a way to globally (or per file) silence specific issues that we may never want to see (say file permissions) I will propose it.

AudriusButkevicius commented:
To be honest, a good chunk of them are also noise, that we should just ignore and not have comments for.

rasa commented:
Ok, I removed all the full line comments, and fixed most of the issues in the code. I did add `//skipcq` in a few test files, *at the end of the line*. I also added a few in non-test files, to quiet "weak rand" and "func name differs only in case" issues.

If you want me to remove these comments too, lemme know if everywhere, or just in non-test files. Thanks.

calmh commented on lib/fs/basicfs_test.go:
```diff
@@ -33,15 +33,15 @@ func TestChmodFile(t *testing.T) {
 	defer os.RemoveAll(dir)
 
-	defer os.Chmod(path, 0666)
+	defer os.Chmod(path, 0666) //skipcq
```
Rather than adding these I think we should just tell deepsource to ignore lax permissions in test files

AudriusButkevicius replied:
Or move the permissions value to a variable

rasa replied:
> For stub methods it's obvious that the parameters are unused, because it's a method stub without a body. Having the parameter names makes the code more natural and readable.

Agreed, but is `//skipcq` acceptable to quiet these? In test files only, or all files?

> For Chtimes in fakefs for example I _don't_ like the change because "_ time.Time, mtime time.Time` just tells me there's some other time passed that we apparently don't care about, but leaves it a mystery what that time is and why we ignore it.

Understood. Restore `atime` but add `//skipcq` to quiet?

> Rather than adding these I think we should just tell deepsource to ignore lax permissions in test files

Sorry, but how? DeepSource's docs show quieting specific lines of code, or ignoring entire files. I don't see a way to ignore specific errors for entire files. I've asked DeepSource if this is possible on their Slack #ideas channel.

> Or move the permissions value to a variable.

Good idea. We'll see if that works.

Note that our [contributing page](https://docs.syncthing.net/dev/contributing.html#tests) may need to be tweaked a bit, as it currently says `That the output from go lint and go vet is clean.` Based on your feedback, I can submit a PR to freshen this statement to clarify what is being asked of our developers.

calmh commented on lib/fs/basicfs_watch_test.go:
```diff
@@ -126,17 +126,17 @@ func TestWatchRename(t *testing.T) {
 
 	old := createTestFile(name, "oldfile")
-	new := "newfile"
+	new_ := "newfile"
```
Underscores are ugly; if we don't want to use "new" then something like "newName", "newFile" or whatever, please.

rasa replied:
Agreed, thx.

calmh reviewed (commented):
Replacing parameter names with underscores makes the code worse, in my opinion, except in a couple of specific exceptions.

For stub methods it's obvious that the parameters are unused, because it's a method stub without a body. Having the parameter names makes the code more natural and readable.

For methods implementing an interface the signature is decided by the interface and it *might* be useful to underscore a parameter to clearly indicate it's unused. Like, a ServeHTTP method that doesn't touch the request might have a `ServeHTTP(_ *http.Request, ...` because it's super clear what the parameter is and it might be useful info that we never touch it.

For Chtimes in fakefs for example I *don't* like the change because "_ time.Time, mtime time.Time` just tells me there's some other time passed that we apparently don't care about, but leaves it a mystery what that time is and why we ignore it. 

Guiding principle: we're writing code for humans to read -- Deepsource doesn't care, and it's wrong.

rasa commented:
Closing due to age. Please reopen if needed.
